### ISSUE TITLE
Failed to create shard cluster on windows

### ISSUE THREAD

**Issue by zhaoyi0113** [rueckstiess/mtools#485]
I am using window10 pro 64bit OS. And I installed `ubuntu bash` on the system.  But`mlaunch` is struck when I creating a shard cluster on windows batch. Below is the command and output.

```
mlaunch --replicaset --sharded 1 --dir /tmp/shard1 --port 28017
launching: mongod on port 28018
launching: mongod on port 28019
launching: mongod on port 28020
launching: config server on port 28021
replica set 'configRepl' initialized.
replica set 'shard01' initialized.
```
It works great without `--sharded 1` parameter. It seems that mongos cluster can't be created.

**Comment by kevinadi** [comment]
@zhaoyi0113 could you post the console output of the command with `--verbose`, e.g.:

```
mlaunch --replicaset --sharded 1 --dir /tmp/shard1 --port 28017
```

Officially mtools does not really support Windows, but hopefully the verbose output can point out what's going on.

**Comment by zhaoyi0113** [comment]
Thanks for your reply. I know `mlaunch` doesn't support on windows but the way I am using it really in a linux way, because I installed `ubuntu bash` on windows.

 Below is the output from `--verbose`. And the process stuck there.

```
# mlaunch --replicaset --sharded 1 --dir /tmp/shard1 --port 28017 --verbose
Detected mongod version: 3.4.5
creating directory: /tmp/shard1/shard01/rs1/db
creating directory: /tmp/shard1/shard01/rs2/db
creating directory: /tmp/shard1/shard01/rs3/db
creating directory: /tmp/shard1/configRepl/rs1/db
waiting for nodes to shutdown...
creating cluster without auth for setup, will enable auth at the end...
launching: mongod --replSet shard01 --dbpath /tmp/shard1/shard01/rs1/db --logpath /tmp/shard1/shard01/rs1/mongod.log --port 28018 --logappend --fork  --shardsvr
launching: mongod --replSet shard01 --dbpath /tmp/shard1/shard01/rs2/db --logpath /tmp/shard1/shard01/rs2/mongod.log --port 28019 --logappend --fork  --shardsvr
launching: mongod --replSet shard01 --dbpath /tmp/shard1/shard01/rs3/db --logpath /tmp/shard1/shard01/rs3/mongod.log --port 28020 --logappend --fork  --shardsvr
launching: mongod --replSet configRepl --dbpath /tmp/shard1/configRepl/rs1/db --logpath /tmp/shard1/configRepl/rs1/mongod.log --port 28021 --logappend --fork  --configsvr
waiting for nodes to start...
Initiating config server replica set.
initializing replica set 'configRepl' with configuration: {'_id': 'configRepl', 'members': [{'host': 'DESKTOP-1FQTLTI:28021', '_id': 0}]}
replica set 'configRepl' initialized.
Initiating shard replica set shard01.
initializing replica set 'shard01' with configuration: {'_id': 'shard01', 'members': [{'host': 'DESKTOP-1FQTLTI:28018', '_id': 0}, {'host': 'DESKTOP-1FQTLTI:28019', '_id': 1}, {'host': 'DESKTOP-1FQTLTI:28020', '_id': 2}]}
replica set 'shard01' initialized.
creating cluster without auth for setup, will enable auth at the end...
```

I open another window to check the `mongod` and `mongos` processes and it seems some of them started:

```
# ps -ef|grep mongo
root      6605     1  1 15:09 ?        00:00:00 mongod --replSet shard01 --dbpath /tmp/shard1/shard01/rs1/db --logpath /tmp/shard1/shard01/rs1/mongod.log --port 28018 --logappend --fork --shardsvr
root      6633     1  1 15:09 ?        00:00:00 mongod --replSet shard01 --dbpath /tmp/shard1/shard01/rs2/db --logpath /tmp/shard1/shard01/rs2/mongod.log --port 28019 --logappend --fork --shardsvr
root      6662     1  0 15:09 ?        00:00:00 mongod --replSet shard01 --dbpath /tmp/shard1/shard01/rs3/db --logpath /tmp/shard1/shard01/rs3/mongod.log --port 28020 --logappend --fork --shardsvr
root      6690     1  1 15:09 ?        00:00:00 mongod --replSet configRepl --dbpath /tmp/shard1/configRepl/rs1/db --logpath /tmp/shard1/configRepl/rs1/mongod.log --port 28021 --logappend --fork --configsvr
root      6847  6564  0 15:09 tty1     00:00:00 /bin/sh -c mongos --logpath /tmp/shard1/mongos.log --port 28017 --configdb configRepl/DESKTOP-1FQTLTI:28021 --logappend   --fork
root      6848  6847  0 15:09 tty1     00:00:00 mongos --logpath /tmp/shard1/mongos.log --port 28017 --configdb configRepl/DESKTOP-1FQTLTI:28021 --logappend --fork
root      6849  6848  0 15:09 ?        00:00:00 mongos --logpath /tmp/shard1/mongos.log --port 28017 --configdb configRepl/DESKTOP-1FQTLTI:28021 --logappend --fork
root      6850  6849  0 15:09 ?        00:00:00 mongos --logpath /tmp/shard1/mongos.log --port 28017 --configdb configRepl/DESKTOP-1FQTLTI:28021 --logappend --fork
root      6984  6973  0 15:10 tty2     00:00:00 grep --color=auto mongo
```

it seems that the mlaunch process is waiting for some configuration.

**Comment by kevinadi** [comment]
Unfortunately we cannot find the exact cause of this yet, since `mtools` was untested in this specific environment. It may be that Windows 10 doesn't emulate the Unix environment as precisely as required by Python, so a specific workaround may be required (if it's possible).

It'll be very helpful if you can provide any details regarding where in the code does `mlaunch` hangs. Of course, pull requests are also more than welcome.

**Comment by stennie** [comment]
@zhaoyi0113 Will this be resolved by your #488 PR?

Cheers,
Stennie

**Comment by zhaoyi0113** [comment]
Yes it will be resolved.
